### ISSUE TITLE
python 3.6 to 3.7 transition

### ISSUE THREAD

**Issue by sigbjorn** [statkraft/shyft#337]
# Overview

python 3.7 is the new one, so we need to prepare for it. 
 The py_37 branch verifies it works with shyft.

At the same time, we need to keep the python 3.6.x version as the master for some time, so that systems in production (that is conda-repositories) supports all major packages we are using.

A quick investigation shows that it's possible to auto-configure these, with a little bit clever scripting, especially in the area of CMake boost -python (not a clean cut in the first place), and the python environment.

**Comment by runeyosh** [comment]
Will wait for Statkraft Anaconda server to upgrade libs to 3.7 (Øystein Torget w/team)

**Comment by oyse** [comment]
We are working on changing the channels on the Anaconda Server to a more sensible structure. Once that is done we can sync the 3.7 packages as well.

**Comment by jfburkhart** [comment]
Will there be backward compatibility with py36? This could be troublesome at UiO.. .

**Comment by sigbjorn** [comment]
Yes, we will have to provide support for 3.6 for quite some time. Just now we are holding back standard build to py 3.6, and providing a branch for 3.7. When time, I will make sure that when building shyft, it builds on py 3.6 or 3.7, depending on the installed/provided version at build time.
